tablib/core.py

```diff
@@ -452,6 +482,164 @@ def load(self, in_stream, format=None, **kwargs):
         import_set(self, in_stream, **kwargs)
         return self
 
+    def _make_format(self, bg_color, font, font_size, font_color, border, bold, italic, aligment):
+        """
+        Creating format for the cell
+        :param bg_color: cell backgroud color, str
+        :param font: font name, str
+        :param font_size: font size, int
+        :param font_color: fonr color, str
+        :param border: border index, int
+        :param bold: bool
+        :param italic: bool
+        :param aligment: text position in the cell, str
+        :return: dict
+        """
+        return {
+            'bg_color': bg_color,
+            'font': font,
+            'font_size': font_size,
+            'font_color': font_color,
+            'border': border,
+            'bold': bold,
+            'italic': italic,
+            'aligment': aligment,
+        }
+
+    def _make_validation(self, first_row, first_col, last_row, last_col, options):
+        """
+        Creating validation for data_validation and condition_format methods
+        :param first_row: int
+        :param first_col: int
+        :param last_row: int
+        :param last_col: int
+        :param options: dict
+        :return: dict
+        """
+        return {
+            'first_row': first_row,
+            'first_col': first_col,
+            'last_row': last_row,
```

Comment:
Do we really have to store last_row and last_col ?
My concern here is that those change as the sheet grows.

Reply:
yes, we need those values in data_validation and condition_format methods:
https://xlsxwriter.readthedocs.io/worksheet.html#data_validation
https://xlsxwriter.readthedocs.io/worksheet.html#conditional_format
because we are working with the range of the cells